packages/components/src/menu-item/index.js

```diff
@@ -74,15 +74,21 @@ export function MenuItem( {
 		props.icon = icon;
 	}
 
+	const atts = {
+		'aria-label': label,
+		role,
+		className,
+		...props,
+	};
+
+	// Make sure aria-checked matches spec https://www.w3.org/TR/wai-aria-1.1/#aria-checked
+	if ( role === 'menuitemcheckbox' || role === 'menuitemradio' ) {
```

Comment:
I've worked on this before and wondered what the right behaviour should be.

Quoting the original issue:
> The component should make sure aria-checked is set only when the role prop is menuitemcheckbox or menuitemradio.

This PR satisfies the above logic.

> Also, it should make sure that when a isSelected prop is passed, the actual intent of the developers is to communicate a "checked" state, thus a menuitemcheckbox or menuitemradio should be used.

However I think this request in the issue still has a question mark—a developer could still easily use the wrong role, only now they'd not have aria-checked output at all.

This PR does at least document the constraint in the form of code, so that's good. I think it'd be good to update the README.md for the MenuItem detailing the intended behaviour of the props as well.